Diagnostics landing (#13744)
dotnet/docs: docs/core/diagnostics/managed-debuggers.md

---

> title: .NET Core Managed Debuggers

Reviewer: ```suggestion
title: Managed debuggers - .NET Core
```

> description: An overview of the Visual Studio and VS Code managed debuggers.

Reviewer: ```suggestion
description: An overview of the Visual Studio and Visual Studio Code managed debuggers.
```

author: sdmaclea
ms.author: stmaclea
ms.date: 08/05/2019
---

> # .NET Core Managed Debuggers

Reviewer: ```suggestion
# .NET Core managed debuggers
```

Debuggers allow programs to be paused or executed step-by-step. When paused, the current state of the process can be viewed. By stepping through key sections, you gain understanding of your code and why it produces the result that it does.

> Microsoft provides debuggers for managed code in the **Visual Studio** and **Visual Studio Code** integrated development environments (IDE).

Reviewer: ```suggestion
Microsoft provides debuggers for managed code in **Visual Studio** and **Visual Studio Code**.
```

Reviewer: What about Visual Studio for Mac?

> ## Visual Studio Managed Debugger

Reviewer: ```suggestion
## Visual Studio managed debugger
```

> **Visual Studio** is an IDE with the most comprehensive debugger available. Visual Studio is an excellent choice for developers working on Windows.

Reviewer: ```suggestion
**Visual Studio** is an integrated development environment with the most comprehensive debugger available. Visual Studio is an excellent choice for developers working on Windows.
```

- [Tutorial - Debugging a .NET Core application on Windows with Visual Studio](../tutorials/debugging-with-visual-studio)

> While Visual Studio is a Windows application, it can still be used to debug Linux and MacOS apps remotely.

Reviewer: ```suggestion
While Visual Studio is a Windows application, it can still be used to debug Linux and macOS apps remotely.

```

> - [Debugging a .Net Core application on Linux/OSX with Visual Studio](https://github.com/Microsoft/MIEngine/wiki/Offroad-Debugging-of-.NET-Core-on-Linux---OSX-from-Visual-Studio)

Reviewer: ```suggestion
- [Debugging a .NET Core application on Linux/OSX with Visual Studio](https://github.com/Microsoft/MIEngine/wiki/Offroad-Debugging-of-.NET-Core-on-Linux---OSX-from-Visual-Studio)
```

 Debugging ASP.NET Core apps require slightly different instructions.

- [Debug ASP.NET Core apps in Visual Studio](/visualstudio/debugger/how-to-enable-debugging-for-aspnet-applications#debug-aspnet-core-apps)

> ## Visual Studio Code Managed Debugger

Reviewer: ```suggestion
## Visual Studio Code managed debugger
```

> **Visual Studio Code** is a lightweight cross-platform code editor. It uses the same .NET Core debugger implementation as Visual Studio, but with a simplified user interface.

Reviewer: ```suggestion
**Visual Studio Code** is a lightweight cross-platform code editor. It uses the same .NET Core debugger implementation as Visual Studio, but with a simplified user interface.

```

> - [Tutorial - Debugging a .Net Core application with Visual Studio Code](../tutorials/with-visual-studio-code#debug)

Reviewer: ```suggestion
- [Tutorial - Debugging a .NET Core application with Visual Studio Code](../tutorials/with-visual-studio-code.md#debug)
```

> - [Visual Studio Code's Debugging Overview](https://code.visualstudio.com/docs/editor/debugging)

Reviewer: ```suggestion
- [Debugging in Visual Studio Code](https://code.visualstudio.com/docs/editor/debugging)
```

---
title: .NET Core Managed Debuggers
description: An overview of the Visual Studio and VS Code managed debuggers.
author: stmaclea
ms.author: stmaclea

> ms.date: 8/5/2019

Reviewer 2: ```suggestion
ms.date: 08/05/2019
```

ms.topic: article
ms.prod: .net-core
---
# .NET Core Managed Debuggers

Debuggers allow programs to be paused or executed step-by-step. When paused, the current state of the process can be viewed. By stepping through key sections, you gain understanding of your code and why it produces the result that it does.

> Microsoft provides debuggers for managed code in the **Visual Studio** and **Visual Studio Code** integrated development environments.

Reviewer 2: ```suggestion
Microsoft provides debuggers for managed code in the **Visual Studio** and **Visual Studio Code** integrated development environments (IDE).
```

## Visual Studio Managed Debugger

> **Visual Studio** is an IDE with the most comprehensive debugger available. Visual Studio is an excellent default choice for developers working on Windows.

Reviewer 2: I don't think **default** is really needed.

```suggestion
**Visual Studio** is an IDE with the most comprehensive debugger available. Visual Studio is an excellent choice for developers working on Windows.
```

>   - [Tutorial - Debugging a .NET Core application on Windows with Visual Studio](https://docs.microsoft.com/dotnet/core/tutorials/debugging-with-visual-studio)

Reviewer 2: Links should be relative to the current article and point to source files unless it points to a different repo.

```suggestion
  - [Tutorial - Debugging a .NET Core application on Windows with Visual Studio](../debugging-with-visual-studio.md)
```

While Visual Studio is a Windows application, it can still be used to debug Linux and MacOS apps remotely.
  - [Debugging a .Net Core application on Linux/OSX with Visual Studio](https://github.com/Microsoft/MIEngine/wiki/Offroad-Debugging-of-.NET-Core-on-Linux---OSX-from-Visual-Studio)

 Debugging ASP.NET Core apps require slightly different instructions.

>   - [Debug ASP.NET Core apps in Visual Studio](https://docs.microsoft.com/visualstudio/debugger/how-to-enable-debugging-for-aspnet-applications#debug-aspnet-core-apps)

Reviewer 2: ```suggestion
  - [Debug ASP.NET Core apps in Visual Studio](how-to-enable-debugging-for-aspnet-applications.md#debug-aspnet-core-apps)
```

Author: This suggestion looks wrong.

Reviewer 2: Sorry, I saw the debugger folder and misread that as the current folder. Instead use `/visualstudio/....the-rest-of-your-link`

>   - [Debug ASP.NET Core apps in Visual Studio](https://docs.microsoft.com/visualstudio/debugger/how-to-enable-debugging-for-aspnet-applications#debug-aspnet-core-apps)

Reviewer 2: ```suggestion
  - [Debug ASP.NET Core apps in Visual Studio](/visualstudio/debugger/how-to-enable-debugging-for-aspnet-applications#debug-aspnet-core-apps)
```

Sorry, I didn't notice this link was pointing to visual studio.

## Visual Studio Code Managed Debugger

> **Visual Studio Code** is a lightweight fully cross platform code editor. It uses the same .NET Core debugger implementation as Visual Studio with a simplified user interface.

Reviewer 2: ```suggestion
**Visual Studio Code** is a lightweight cross-platform code editor. It uses the same .NET Core debugger implementation as Visual Studio, but with a simplified user interface.
```
